packages/mui-base/src/utils/BaseUI.types.ts

```diff
@@ -1,3 +1,19 @@
+export type BaseUIEvent<E extends React.SyntheticEvent<Element, Event>> = E & {
+  preventBaseUIHandler: () => void;
+};
+
+type WithPreventBaseUIHandler<T> = T extends (event: infer E) => any
+  ? E extends React.SyntheticEvent<Element, Event>
+    ? (event: BaseUIEvent<E>) => ReturnType<T>
+    : never
+  : T extends undefined
+    ? undefined
+    : T;
+
+export type BaseUIProps<T> = {
```

Comment:
I know naming is hard :) `BaseUIProps` and `BaseUIComponentProps` are too similar and just by looking at their names it's hard to figure out what is responsible for what.